backends/ignite.go

```diff
@@ -0,0 +1,256 @@
+			errMsg := fmt.Sprintf("Error creating Ignite backend: %s", err.Error())
+			log.Fatalf(errMsg)
+			panic(errMsg)
+		}
+	}
+	log.Infof("Prebid Cache will write to Ignite cache name: %s", cfg.Cache.Name)
+
+	return igb
+}
+
+// createCache uses the Apache Ignite REST API "getorcreate" command to create a cache
+func createCache(igb *IgniteBackend) error {
+
+	urlCopy := *igb.serverURL
+	q := urlCopy.Query()
+	q.Set("cmd", "getorcreate")
+	q.Set("cacheName", igb.cacheName)
+	urlCopy.RawQuery = q.Encode()
+
+	responseBytes, err := igb.sender.DoRequest(context.Background(), &urlCopy, nil)
+	if err != nil {
+		return err
+	}
+
+	igniteResponse := getResponse{}
+
+	if unmarshalErr := json.Unmarshal(responseBytes, &igniteResponse); unmarshalErr != nil {
+		return fmt.Errorf("Unmarshal response error: %s; Response body: %s", unmarshalErr.Error(), string(responseBytes))
+	}
+
+	if len(igniteResponse.Error) > 0 {
+		return fmt.Errorf("Ignite error. %s", igniteResponse.Error)
+	}
+	if igniteResponse.Status > 0 {
+		return fmt.Errorf("Ignite error. successStatus does not equal 0 %v", igniteResponse)
+	}
+
+	return nil
+}
+
+// getResponse is used to unmarshal the Ignite server's response to a GET request with
+// the "cmd" URL query field set to "get"
+type getResponse struct {
+	Error    string `json:"error"`
+	Response string `json:"response"`
+	Status   int    `json:"successStatus"`
+}
+
+// Get implements the Backend interface. Makes the Ignite storage client retrieve the value that has
+// been previously stored under 'key' if its TTL is still current. We can tell when a key is not found
+// when Ignite doesn't return an error, nor a 'Status' different than zero, but the 'Response' field is
+// empty. Get can also return Ignite server-side errors
+func (ig *IgniteBackend) Get(ctx context.Context, key string) (string, error) {
+	urlCopy := *ig.serverURL
+	q := urlCopy.Query()
+	q.Set("cmd", "get")
+	q.Set("key", key)
+
+	urlCopy.RawQuery = q.Encode()
+
+	responseBytes, err := ig.sender.DoRequest(ctx, &urlCopy, ig.headers)
+	if err != nil {
+		return "", err
+	}
+
+	// Unmarshal response
+	igniteResponse := getResponse{}
+
+	if unmarshalErr := json.Unmarshal(responseBytes, &igniteResponse); unmarshalErr != nil {
+		return "", fmt.Errorf("Unmarshal response error: %s; Response body: %s", unmarshalErr.Error(), string(responseBytes))
+	}
+
+	// Validate response
+	if len(igniteResponse.Error) > 0 {
+		return "", utils.NewPBCError(utils.GET_INTERNAL_SERVER, igniteResponse.Error)
+	} else if igniteResponse.Status > 0 {
+		return "", utils.NewPBCError(utils.GET_INTERNAL_SERVER, "Ignite response.Status not zero")
+	} else if len(igniteResponse.Response) == 0 {
+		return "", utils.NewPBCError(utils.KEY_NOT_FOUND)
+	}
+
+	return igniteResponse.Response, nil
+}
+
+// putResponse is used to unmarshal the Ignite server's response to a PUT request with
+// the "cmd" URL query field set to "putifabs"
+type putResponse struct {
```

Comment:
Is there a reason for defining both a `getResponse` and a `putResponse` when they both seem to be the same structure?

Reply:
The difference is that in `putResponse.Response` is a `bool` and `getResponse.Response` is a `string`. Do you want me to try to consolidate them into a single `struct`?